libutils/string_lib.c

```diff
@@ -77,7 +77,7 @@ unsigned int StringHash(const char *str, unsigned int seed, unsigned int max)
     h ^= (h >> 11);
     h += (h << 15);
 
-    return (h & (max - 1));
+    return (h % max);
```

Comment:
Alright, glad this is resolved now, these are subtle issues! Feel free to close this issue if there is nothing to fix, or maybe submit something different.

FWIW I prefer to have ASSERT(power-of-two) plus binary AND operation, rather than one-size-fits-all modulo operation. The reason is that modulo skews results and creates uneven distribution in subtle cases. But ASSERT works as documentation, enforcing proper usage.